### DISCUSSION TITLE
Compressed saving of networks

### DISCUSSION THREAD

**Discussion by dlohmeier** [e2nIEE/pandapower#1300] [Ideas]
I believe that this has already been discussed a few times, but it seems to me that compression of pandapower networks has not yet been implemented yet, so each user has to find his own solution. In my case, I tried to save a JSON string with gzip and it worked perfectly, compressing the 550 KB of initial data to 60 KB. That makes my studies a lot easier, in which I need to save different network configurations as results many times. What do you think about the idea of implementing a new way to store networks? Here are my code snippets:

    def to_gzip(net, filename, compresslevel=9):
        net_str = pp.to_json(net).encode("utf-8")
        with gzip.open(filename, "wb", compresslevel=compresslevel) as fl:
            fl.write(net_str)
    
    
    def from_gzip(filename):
        with gzip.open(filename, 'rb') as fl:
            net_str = fl.read()
            return pp.from_json_string(net_str)

Of course, this would increase the number of functions included in pandapower, but I think it could be really useful in some cases.

**Comment by MacsBO** [comment]
I agree, I also wrote some functions to compress json and save a lot of disk space: having a standard pandapower routine would be perfect!
